x-pack/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/span_flyout/span_db.tsx

```diff
@@ -5,65 +5,20 @@
  * 2.0.
  */
 
-import { EuiSpacer, EuiTitle } from '@elastic/eui';
+import { EuiSpacer, EuiTitle, EuiCodeBlock } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
-import { tint } from 'polished';
 import React, { Fragment } from 'react';
-import { Light as SyntaxHighlighter } from 'react-syntax-highlighter';
-import sql from 'react-syntax-highlighter/dist/cjs/languages/hljs/sql';
-import xcode from 'react-syntax-highlighter/dist/cjs/styles/hljs/xcode';
-import { euiStyled } from '../../../../../../../../../../../src/plugins/kibana_react/common';
 import { Span } from '../../../../../../../../typings/es_schemas/ui/span';
-import { useTheme } from '../../../../../../../hooks/use_theme';
-import { TruncateHeightSection } from './truncate_height_section';
-
-SyntaxHighlighter.registerLanguage('sql', sql);
-
-const DatabaseStatement = euiStyled.div`
-  padding: ${({ theme }) =>
-    `${theme.eui.paddingSizes.s} ${theme.eui.paddingSizes.m}`};
-  background: ${({ theme }) => tint(0.9, theme.eui.euiColorWarning)};
-  border-radius: ${({ theme }) => theme.eui.euiBorderRadiusSmall};
-  border: 1px solid ${({ theme }) => theme.eui.euiColorLightShade};
-  font-family: ${({ theme }) => theme.eui.euiCodeFontFamily};
-  font-size: ${({ theme }) => theme.eui.euiFontSizeS};
-`;
 
 interface Props {
   spanDb?: NonNullable<Span['span']>['db'];
 }
 
 export function SpanDatabase({ spanDb }: Props) {
-  const theme = useTheme();
-  const dbSyntaxLineHeight = theme.eui.euiSizeL;
-  const previewHeight = 240; // 10 * dbSyntaxLineHeight
-
   if (!spanDb || !spanDb.statement) {
     return null;
   }
 
-  const statementItem =
-    spanDb.type !== 'sql' ? (
-      spanDb.statement
-    ) : (
-      <TruncateHeightSection previewHeight={previewHeight}>
```

Comment:
I think this component is kind of important to present a better UI. If we have a long query it will take a huge amount of space on the screen. Do you mind testing it out?

Reply:
It looks like `EuiCode` supports `overflowHeight` in the [docs](https://eui.elastic.co/#/editors-syntax/code) - might be a better option for handling those large displays of code.

```
<EuiCodeBlock language="sql" overflowHeight={150}>
    {sqlCode}
  </EuiCodeBlock>
```